prov/util/src/util_mr_map.c

```diff
@@ -301,3 +303,23 @@ int ofi_mr_verify(struct ofi_mr_map *map, ssize_t len,
 	fastlock_release(&domain->lock);
 	return ret;
 }
+
+enum fi_hmem_iface ofi_get_mr_device_iface(struct ofi_mr_map *map, void **desc,
+					   uint64_t *device)
+{
+	struct util_domain *domain;
+	struct ofi_mr *mr;
+
+	domain = container_of(map, struct util_domain, mr_map);
+	if (!(domain->mr_mode & FI_MR_HMEM) || !desc || !*desc)
```

Comment:
Should the FI_HMEM capability be checked instead of FI_MR_HMEM? I think FI_HMEM can be supported without the FI_MR_HMEM mode bit.

Reply:
Can it? I'm not seeing how, exactly. If HMEM memory is not registered, how is the provider supposed to know what interface to use?

Reply:
In theory, a provider could support all interfaces without requiring FI_MR_HMEM by checking the buffer with each interface (say, with `cuGetPointerAttributes` for CUDA). I can't imagine a sane provider doing this on each transfer though.

Reply:
I thought that was the idea behind having the hooking providers for HMEM interfaces (I'm working on the ZE one). So a provider can assume/require MR registration from the app and know the interface but the hook would have to deal with all the insanity of figuring out what type of memory it is and registering it appropriately.

Reply:
I know, I was just giving an example of a hypothetical case. Having a hook per interface make total sense to existing enable middleware without any FI_HMEM modifications.

Reply:
Ok cool. So do you think it's safe to assume if the provider is not requiring MR_HMEM registration, that we can return HMEM_SYSTEM? Or to account for the hypothetical case, we should query all interfaces....?

Reply:
While a core provider might not query all interfaces types, it could choose to do for one interface what a hook might do, and this might cause a problem then. I have no strong feelings though.